lib/src/data_stream/stream_writer.dart

```diff
@@ -83,26 +80,32 @@ class StreamWriterImpl implements StreamWriter<String> {
         streamChunk: chunk,
       );
       await engine.sendDataPacket(chunkPacket, reliability: true);
-
       chunkId += 1;
     }
   }
-}
 
-List<Uint8List> splitUtf8(String text, int chunkSize) {
-  final utf8Bytes = utf8.encode(text);
-  final chunks = <Uint8List>[];
-  if (text.length <= chunkSize) {
-    chunks.add(Uint8List.fromList(utf8Bytes));
-    return chunks;
-  }
-  for (var i = 0; i < utf8Bytes.length; i += chunkSize) {
-    final end = i + chunkSize;
-    if (end > utf8Bytes.length) {
-      chunks.add(Uint8List.fromList(utf8Bytes.sublist(i)));
-      break;
+  List<Uint8List> splitUTF8List(T chunk, int chunkSize) {
```

Comment:
this method doesn't actually split at utf8 characters. 

there's an edge case where a multi byte utf-8 character (e.g. an emoji) is at the exact end of the stream chunk length.

So this multi-byte character would get split in the middle. I think we still need to avoid that. See https://github.com/livekit/client-sdk-js/blob/main/src/room/utils.ts#L629-L652

Reply:
thanks for the tip and solution. just resolved and passed the unit test for multi-byte string。